### ISSUE TITLE
Include `reason` label to the `kube_deployment_status_condition` metric

### ISSUE THREAD

**Issue by augustfengd** [kubernetes/kube-state-metrics#1991]
**What would you like to be added**:

I would like the label `reason` to be added to the `kube_deployment_status_condition` metric.

**Why is this needed**:

The `kube_deployment_status_condition` has only two properties relating to the `DeploymentCondition` object: `condition` and `status`, which is not enough to distinguish between a [Progressing Deployment](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#progressing-deployment) and a [Complete Deployment](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#complete-deployment). The `reason` label will add sufficient information to distinguish the two states.

**Describe the solution you'd like**

Simply extract and emit the `reason` label for the deployments.

**Additional context**

.

**Comment by CatherineF-dev** [comment]
However, it's a stable metric stated in https://github.com/kubernetes/kube-state-metrics/blob/main/docs/deployment-metrics.md

Any ideas for this case? Should we introduce a new metric? cc @dgrisonnet

**Comment by CatherineF-dev** [comment]
Got it. 
Created https://github.com/kubernetes/kube-state-metrics/issues/1995 and https://github.com/kubernetes/kube-state-metrics/pull/1996 to discuss definitions of `stable metrics` in kube-state-metrics repo.

**Comment by logicalhan** [comment]
> Adding a new label is not a breaking change, so as long as its values are bounded, I would be fine with adding it.

It depends on your ingestion. Since people ingest metrics into stores other than prometheus, I would argue that the lowest common denominator dictates that it is a breaking change.

**Comment by CatherineF-dev** [comment]
TLDR: we can add new labels for KSM stable metric.

cc @augustfengd  Feel free to add a new label if you're available.

**Comment by A-Hilaly** [comment]
Anyone working on this? If none i'd like to give it a shot :)

**Comment by Gopinath-Selvam** [comment]
Anyone working on this? If not, then I'd like to contribute for this!

**Comment by opeco17** [comment]
Since there is no update for a long time, I created PR (https://github.com/kubernetes/kube-state-metrics/pull/2146).

**Comment by dgrisonnet** [comment]
@opeco17 would you be interested in reviving your PR?

Otherwise would @a-hilaly or @Gopinath-Selvam be interested in taking over Opeco's PR and address the pending comments?

/triage accepted

**Comment by Rishab87** [comment]
seems like this was not resolved, can I take upon the existing work and resolve the comments @dgrisonnet?

**Comment by Rishab87** [comment]
following my previous comment I made a PR for it since it was inactive from past 2 years, I addressed the reviews and added allow list along with a test for it. Would really appreciate if you can have a look.
